src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1220,7 +1226,7 @@ public function getHost()
         // check that it does not contain forbidden characters (see RFC 952 and RFC 2181)
         // use preg_replace() instead of preg_match() to prevent DoS attacks with long host names
         if ($host && '' !== preg_replace('/(?:^\[)?[a-zA-Z0-9-:\]_]+\.?/', '', $host)) {
-            throw new \UnexpectedValueException(sprintf('Invalid Host "%s"', $host));
+            throw new InvalidHostException($host, sprintf('Invalid host "%s".', $host));
```

Comment:
Couldn't we just pass the message as it contains the host, such as done for `UntrustedHostException`?